drivers/bridge/setup_ip_tables.go

```diff
@@ -29,17 +70,18 @@ func (n *bridgeNetwork) setupIPTables(config *networkConfiguration, i *bridgeInt
 		return fmt.Errorf("Failed to Setup IP tables: %s", err.Error())
 	}
 
-	_, err = iptables.NewChain(DockerChain, config.BridgeName, iptables.Nat, hairpinMode)
+	_, filterChain, err := n.getDriverChains()
 	if err != nil {
-		return fmt.Errorf("Failed to create NAT chain: %s", err.Error())
+		return fmt.Errorf("Failed to setup IP tables, cannot acquire chain info %s", err.Error())
 	}
+	filterChain.Bridge = config.BridgeName
```

Comment:
Yes this is very good point. There are different ways to address this. Please see if the patch I am uploading shortly is reasonable.